pkg/hook/hook.go

```diff
@@ -123,16 +128,32 @@ func (r *HookRunner) Run(ctx context.Context) {
 			if err := r.hook.Do(ctx, hash); err != nil {
 				r.logger.Error(err, "hook failed")
 				updateHookRunCountMetric(r.hook.Name(), "error")
+				// don't want to sleep unnecessarily terminating anyways
+				r.sendCompletedOnceMessageIfApplicable(false)
 				time.Sleep(r.backoff)
```

Comment:
If we do send a notification, it means the parent will exit soon, so we shouldn't even bother to retry, or we end up with anaother race, right?  If we pass `oneTime bool` into the hook creation, then we know not to bother here, and we can be smart enough to return an error if you like my idea of a method above.  E.g.

```
if !r.oneTime {
    return fmt.Errorf("hook.WaitForCompletion called on async runner")
}